### PR TITLE
Add test for first checkpoint verification

### PR DESCRIPTION
**Pull request by yaahc**

### CODE REVIEW

**Review by oxarbitrage — DISMISSED** [review]
This is great. Can be applied to the other tests as well. I think is ok to merge to proposed branch `test-cleaning` as it is and continue there or in a separated pr.

File: zebrad/tests/acceptance.rs
Comment (teor2345) [inline zebrad/tests/acceptance.rs]: I like this new API!

We want to do networking tests for mainnet and testnet, so it would be great to have a function that sets the network based on a zebra_chain::Network value.

Comment (yaahc) [reply zebrad/tests/acceptance.rs]: hmm, i looked around to see where we mess with the Network for tests and it seems to be only ever as part of the config. As such I'm thinking it might be better to rework how we pass around / in a config to make it easier to manipulate the Network. I'll push up a version of what I mean rn.

Comment (teor2345) [reply zebrad/tests/acceptance.rs]: Most of the app just uses the `config.network.network` value to make config-related decisions.

The one exception is `config.network.listener`, where the default port should change for testnet. In this case, we just issue a warning, because using 8233 on the testnet is a strange but legitimate config.

Comment (yaahc) [reply zebrad/tests/acceptance.rs]: sure, but this could all be part of the API on `Config` rather than the API on the test runners.

For example:

```
testdir()?
    .with_config(default_test_config().with_network(Testnet))
    .spawn_child(&["start"])?
    .with_timeout(Duration::from_secs(1))
    .expect_stdout("verified checkpoint range")?
    .kill()
```

then the `.with_network` function could even handle the listener issue you mentioned.

Comment (teor2345) [reply zebrad/tests/acceptance.rs]: Sounds good!

**Comment by teor2345** [comment]
Do you know why the CI trigger is failing here?

Other PRs seem to work fine.

**Comment by dconnolly** [comment]
> Do you know why the CI trigger is failing here?
> 
> Other PRs seem to work fine.

image

**Comment by yaahc** [comment]
oh yea im in the middle of fixing this locally rn

**Comment by teor2345** [comment]
@dconnolly the CI trigger seems to need a login to google to see those details.

All I see is:
All Checks Failed
All Checks Details
All Checks Login

But other failures show up publicly.

Can we show this failure publicly as well?

**Comment by dconnolly** [comment]
> @dconnolly the CI trigger seems to need a login to google to see those details.
> 
> All I see is:
> All Checks Failed
> All Checks Details
> All Checks Login
> 
> But other failures show up publicly.
> 
> Can we show this failure publicly as well?

You have access to these runs by logging in with your @zfnd.org account. 

This integration is via the Google Cloud Build GitHub App, allowing these builds to be triggered on PRs from forks, unlike the Google Cloud GitHub Actions, which have the  api secret issue, and cannot be triggered by PRs from forks. The other jobs (ubuntu/macos/windows) are GitHub Actions and can be triggered on PRs from forks because they don't need any other secrets to run and report their results. We have both the gcloud builds and the other platform builds because the gcloud builds are on a 32core machine and finish the fastest, and are building the same Docker container that we build and deploy on merge to main, but we can't get their console out, so we have the other GitHub Actions jobs to get the console out and test cross-platform compatibility.

**Comment by teor2345** [comment]
> You have access to these runs by logging in with your @zfnd.org account.

Yes, I know 😅 

> This integration is via the Google Cloud Build GitHub App, allowing these builds to be triggered on PRs from forks, unlike the Google Cloud GitHub Actions, which have the api secret issue, and cannot be triggered by PRs from forks. The other jobs (ubuntu/macos/windows) are GitHub Actions and can be triggered on PRs from forks because they don't need any other secrets to run and report their results.

Ok, that's complicated, but understandable.

I was concerned that seeing a single unexplained failure is a bad user experience: 
All Checks Failed

But I think that can only happen to people with write access to ZcashFoundation/zebra, who don't have a logged-in zfnd.org account. So it's not as big a concern as I originally thought.

**Comment by teor2345** [comment]
@dconnolly separate but related question:

Why does this PR show 1 check, but other PRs like #902 show 13 checks?

**Review by teor2345 — CHANGES_REQUESTED** [review]
I think this code looks good.

But the timeout will need to be tuned. We might want to do that after #993 merges. I made a note in #1011.

I'm not sure why our 12 GitHub action CI jobs aren't running on this PR, but I'd really like to see those results before we merge.

**Comment by teor2345** [comment]
I'm going to try closing and re-opening this PR to trigger the GitHub actions.

File: zebrad/tests/acceptance.rs
Comment (teor2345) [inline zebrad/tests/acceptance.rs:511]: I wonder if this timeout is too short - I guess we'll find out after we run the tests some more.

Comment (yaahc) [reply zebrad/tests/acceptance.rs:511]: yolo!

Comment (teor2345) [reply zebrad/tests/acceptance.rs:511]: It seems to work fine on the testnet, see PR #1035.

I also saw a crash due to out of order blocks, but that will be fixed by #1011.

**Review by teor2345 — DISMISSED** [review]
I'd like to merge this branch, but it seems to disable our 12 GitHub action CI jobs.

It's not just a transient GitHub bug in this particular PR - other PRs based on this branch also disable all those jobs. (For example, #1035.)

So the breaking change might be in the code in this branch.

Maybe @dconnolly can help us work out why those tests aren't running?

Note that this PR is based on the test-cleaning branch in PR #1017, which runs all the CI checks. So the breaking change should be visible in this PR.

**Comment by dconnolly** [comment]
OH. This is why:

image

I hadn't noticed this was a PR into #test-cleaning until now

**Comment by yaahc** [comment]
@teor2345 @dconnolly okay I fixed things up now so it should be ready for merge. I had to do some rebasing to remove the old history it was based on from the changes to block commit ordering I squashed everything into one commit, changed the base branch to main, and closed the draft PR that this was based on. There's now more code that's part of this PR so a re-review is probably needed assuming either of you are not already familiar with the changes in `test-cleanup`.

**Review by dconnolly — DISMISSED** [review]

**Comment by dconnolly** [comment]
I also just merged to main that all PRs, including those not targeting #main, will run the CI jobs (except for Coverage which is all relative to #main)

**Review by dconnolly — APPROVED** [review]

**Review by teor2345 — APPROVED** [review]
Looks good to me!

**Pull request merged**
